diff --git a/common/shlibs b/common/shlibs
index d77d393f..76176ec6 100644
--- a/common/shlibs
+++ b/common/shlibs
@@ -3405,3 +3405,10 @@ libbelcard.so.1 belcard-1.0.2_1
 libshaderc_shared.so shaderc-2017.2_1
 libglslang.so glslang-6.2.2596_1
 libSPIRV.so glslang-6.2.2596_1
+libsecp256k1.so.0 libsecp256k1-20180531_1
+libbitcoin.so.0 libbitcoin-system-3.5.0_1
+libbitcoin-consensus.so.0 libbitcoin-consensus-3.5.0_1
+libbitcoin-protocol.so.0 libbitcoin-protocol-3.5.0_1
+libbitcoin-client.so.0 libbitcoin-client-3.5.0_1
+libbitcoin-network.so.0 libbitcoin-network-3.5.0_1
+libbitcoin-explorer.so.0 libbitcoin-explorer-3.5.0_1
diff --git a/srcpkgs/libbitcoin-client-devel b/srcpkgs/libbitcoin-client-devel
new file mode 120000
index 00000000..b52c6b7d
--- /dev/null
+++ b/srcpkgs/libbitcoin-client-devel
@@ -0,0 +1 @@
+libbitcoin-client
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-client/template b/srcpkgs/libbitcoin-client/template
new file mode 100644
index 00000000..38ab872e
--- /dev/null
+++ b/srcpkgs/libbitcoin-client/template
@@ -0,0 +1,42 @@
+# Template file for 'libbitcoin-client'
+pkgname=libbitcoin-client
+version=3.5.0
+revision=1
+build_style=gnu-configure
+configure_args="--with-gnu-ld --without-examples"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost-devel icu-devel libbitcoin-protocol-devel
+ libbitcoin-system-devel libpng-devel libsecp256k1-devel protobuf-devel
+ qrencode-devel zeromq-devel"
+short_desc="Bitcoin Client Query Library"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin-client"
+distfiles="https://codeload.github.com/libbitcoin/$pkgname/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=bafa26647f334ecad04fc4bbef507a1954d7e0682f07bd38b90ab66dba5fe0d2
+shlib_provides="libbitcoin-client.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-client-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
diff --git a/srcpkgs/libbitcoin-consensus-devel b/srcpkgs/libbitcoin-consensus-devel
new file mode 120000
index 00000000..01eb2cda
--- /dev/null
+++ b/srcpkgs/libbitcoin-consensus-devel
@@ -0,0 +1 @@
+libbitcoin-consensus
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-consensus/template b/srcpkgs/libbitcoin-consensus/template
new file mode 100644
index 00000000..6d801b43
--- /dev/null
+++ b/srcpkgs/libbitcoin-consensus/template
@@ -0,0 +1,40 @@
+# Template file for 'libbitcoin-consensus'
+pkgname=libbitcoin-consensus
+version=3.5.0
+revision=1
+build_style=gnu-configure
+configure_args="--with-gnu-ld"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost boost-devel libsecp256k1-devel"
+short_desc="Bitcoin Consensus Library"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin-consensus"
+distfiles="https://codeload.github.com/libbitcoin/$pkgname/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=bb29761d4275a9c993151707557008b23572a3d9adecc0e36a3075cfb101dd1e
+shlib_provides="libbitcoin-consensus.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-consensus-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
diff --git a/srcpkgs/libbitcoin-explorer-devel b/srcpkgs/libbitcoin-explorer-devel
new file mode 120000
index 00000000..f064c373
--- /dev/null
+++ b/srcpkgs/libbitcoin-explorer-devel
@@ -0,0 +1 @@
+libbitcoin-explorer
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-explorer/template b/srcpkgs/libbitcoin-explorer/template
new file mode 100644
index 00000000..87328b29
--- /dev/null
+++ b/srcpkgs/libbitcoin-explorer/template
@@ -0,0 +1,43 @@
+# Template file for 'libbitcoin-explorer'
+pkgname=libbitcoin-explorer
+version=3.5.0
+revision=1
+build_style=gnu-configure
+configure_args="--with-bash-completiondir=/usr/share/bash-completion/completions
+ --with-gnu-ld"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost-devel icu-devel libbitcoin-client-devel
+ libbitcoin-network-devel libbitcoin-protocol-devel libbitcoin-system-devel
+ libpng-devel libsecp256k1-devel protobuf-devel qrencode-devel zeromq-devel"
+short_desc="Bitcoin Command Line Tool"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin-explorer"
+distfiles="https://codeload.github.com/libbitcoin/$pkgname/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=630cffd577c0d10345b44ce8160f4604519b0ca69bf201f524f104c207b930aa
+shlib_provides="libbitcoin-explorer.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-explorer-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
diff --git a/srcpkgs/libbitcoin-network-devel b/srcpkgs/libbitcoin-network-devel
new file mode 120000
index 00000000..c4644114
--- /dev/null
+++ b/srcpkgs/libbitcoin-network-devel
@@ -0,0 +1 @@
+libbitcoin-network
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-network/template b/srcpkgs/libbitcoin-network/template
new file mode 100644
index 00000000..4f278087
--- /dev/null
+++ b/srcpkgs/libbitcoin-network/template
@@ -0,0 +1,41 @@
+# Template file for 'libbitcoin-network'
+pkgname=libbitcoin-network
+version=3.5.0
+revision=1
+build_style=gnu-configure
+configure_args="--with-gnu-ld"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost-devel icu-devel libbitcoin-system-devel libpng-devel
+ libsecp256k1-devel qrencode-devel"
+short_desc="Bitcoin P2P Network Library"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin-network"
+distfiles="https://codeload.github.com/libbitcoin/$pkgname/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=e065bd95f64ad5d7b0f882e8759f6b0f81a5fb08f7e971d80f3592a1b5aa8db4
+shlib_provides="libbitcoin-network.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-network-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
diff --git a/srcpkgs/libbitcoin-protocol-devel b/srcpkgs/libbitcoin-protocol-devel
new file mode 120000
index 00000000..9b775877
--- /dev/null
+++ b/srcpkgs/libbitcoin-protocol-devel
@@ -0,0 +1 @@
+libbitcoin-protocol
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-protocol/template b/srcpkgs/libbitcoin-protocol/template
new file mode 100644
index 00000000..5f4ecfdd
--- /dev/null
+++ b/srcpkgs/libbitcoin-protocol/template
@@ -0,0 +1,41 @@
+# Template file for 'libbitcoin-protocol'
+pkgname=libbitcoin-protocol
+version=3.5.0
+revision=1
+build_style=gnu-configure
+configure_args="--with-gnu-ld"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost-devel icu-devel libbitcoin-system-devel libpng-devel
+ libsecp256k1-devel protobuf-devel qrencode-devel zeromq-devel"
+short_desc="Bitcoin Blockchain Query Protocol"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin-protocol"
+distfiles="https://codeload.github.com/libbitcoin/$pkgname/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=9deac6908489e2d59fb9f89c895c49b00e01902d5fdb661f67d4dbe45b22af76
+shlib_provides="libbitcoin-protocol.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-protocol-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
diff --git a/srcpkgs/libbitcoin-system-devel b/srcpkgs/libbitcoin-system-devel
new file mode 120000
index 00000000..13797e5b
--- /dev/null
+++ b/srcpkgs/libbitcoin-system-devel
@@ -0,0 +1 @@
+libbitcoin-system
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-system/template b/srcpkgs/libbitcoin-system/template
new file mode 100644
index 00000000..128ab10e
--- /dev/null
+++ b/srcpkgs/libbitcoin-system/template
@@ -0,0 +1,43 @@
+# Template file for 'libbitcoin-system'
+pkgname=libbitcoin-system
+version=3.5.0
+revision=1
+build_style=gnu-configure
+configure_args="--with-gnu-ld --with-icu --with-png --with-qrencode
+ --without-examples"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost-devel icu-devel libpng-devel libsecp256k1-devel
+ qrencode-devel"
+short_desc="Bitcoin Cross-Platform C++ Development Toolkit"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin"
+distfiles="https://codeload.github.com/libbitcoin/${pkgname%-system}/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=214d9cd6581330b0e1f6fd8f0c634c46b75ae5515806ecac189f21c0291ae2d9
+wrksrc="${pkgname%-system}-$version"
+shlib_provides="libbitcoin.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-system-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
diff --git a/srcpkgs/libsecp256k1-devel b/srcpkgs/libsecp256k1-devel
new file mode 120000
index 00000000..1a5b8eeb
--- /dev/null
+++ b/srcpkgs/libsecp256k1-devel
@@ -0,0 +1 @@
+libsecp256k1
\ No newline at end of file
diff --git a/srcpkgs/libsecp256k1/template b/srcpkgs/libsecp256k1/template
new file mode 100644
index 00000000..0eacab67
--- /dev/null
+++ b/srcpkgs/libsecp256k1/template
@@ -0,0 +1,38 @@
+# Template file for 'libsecp256k1'
+pkgname=libsecp256k1
+version=20180531
+revision=1
+build_style=gnu-configure
+configure_args="--disable-benchmark --disable-coverage --disable-jni
+ --disable-openssl-tests --enable-exhaustive-tests --enable-module-recovery
+ --enable-tests --with-gnu-ld"
+hostmakedepends="autoconf automake libtool m4 pkg-config"
+short_desc="Optimized C library for EC operations on curve secp256k1"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="MIT"
+homepage="https://github.com/bitcoin-core/secp256k1"
+_gitrev="1e6f1f5ad5e7f1e3ef79313ec02023902bf8175c"
+distfiles="https://codeload.github.com/bitcoin-core/secp256k1/tar.gz/$_gitrev>$pkgname-$version.tar.gz"
+checksum=d4bc033398d4db43077ceb3aa50bb2f7700bdf3fc6eb95b8c799ff6f657a804a
+wrksrc="${pkgname#lib}-$_gitrev"
+shlib_provides="libsecp256k1.so.0"
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libsecp256k1-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
