From c00da7c1e6d85dfd7a29621823dfb836c435e35d Mon Sep 17 00:00:00 2001
From: Andy Weidenbaum <atweiden@ioiojo.com>
Date: Tue, 30 Aug 2022 09:46:43 +1000
Subject: [PATCH] allow cmake -DCMAKE_BUILD_TYPE=None for void

---
 doc/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index 491ed801..ca5cd0a3 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -15,6 +15,5 @@ ELSE (WIN32)
 
 	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/cryfs.1.gz
 			DESTINATION ${CMAKE_INSTALL_MANDIR}/man1
-			CONFIGURATIONS Release
 	)
 ENDIF(WIN32)
-- 
2.37.2

