diff --git a/src/cpp-utils/thread/debugging_nonwindows.cpp b/src/cpp-utils/thread/debugging_nonwindows.cpp
index 3ed543ff..d613a31d 100644
--- a/src/cpp-utils/thread/debugging_nonwindows.cpp
+++ b/src/cpp-utils/thread/debugging_nonwindows.cpp
@@ -28,9 +28,27 @@ void set_thread_name(const char* name) {
 }
 
 namespace {
+// Get the name of a thread.
+int pthread_getname_np_gcompat(pthread_t thread, char *name, size_t len) {
+  int fd = open("/proc/thread-self/comm", O_RDONLY | O_CLOEXEC);
+  char dummy;
+  if (fd < 0)
+    return errno;
+  if (read(fd, name, len) < 0)
+    return errno;
+  // If there's more to read, the buffer was too small.
+  if (read(fd, &dummy, 1) > 0)
+    return ERANGE;
+  return 0;
+}
+
 std::string get_thread_name(pthread_t thread) {
   char name[MAX_NAME_LEN];
+#ifdef __GLIBC__
   int result = pthread_getname_np(thread, name, MAX_NAME_LEN);
+#else
+  int result = pthread_getname_np_gcompat(thread, name, MAX_NAME_LEN);
+#endif
   if (0 != result) {
     throw std::runtime_error("Error getting thread name with pthread_getname_np. Code: " + std::to_string(result));
   }
