diff --git a/common/shlibs b/common/shlibs
index 063a8a0c..4f35e0bb 100644
--- a/common/shlibs
+++ b/common/shlibs
@@ -3411,3 +3411,4 @@ libbitcoin-explorer.so.0 libbitcoin-explorer-3.5.0_1
 libbitcoin-database.so.0 libbitcoin-database-3.5.0_1
 libbitcoin-blockchain.so.0 libbitcoin-blockchain-3.5.0_1
 libbitcoin-node.so.0 libbitcoin-node-3.5.0_1
+libbitcoin-server.so.0 libbitcoin-server-3.5.0_1
diff --git a/srcpkgs/libbitcoin-server-devel b/srcpkgs/libbitcoin-server-devel
new file mode 120000
index 00000000..1ac5e3a4
--- /dev/null
+++ b/srcpkgs/libbitcoin-server-devel
@@ -0,0 +1 @@
+libbitcoin-server
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-server/template b/srcpkgs/libbitcoin-server/template
new file mode 100644
index 00000000..b5559108
--- /dev/null
+++ b/srcpkgs/libbitcoin-server/template
@@ -0,0 +1,45 @@
+# Template file for 'libbitcoin-server'
+pkgname=libbitcoin-server
+version=3.5.0
+revision=1
+only_for_archs="aarch64 aarch64-musl x86_64 x86_64-musl"
+build_style=gnu-configure
+configure_args="--with-bash-completiondir=/usr/share/bash-completion/completions
+ --with-gnu-ld"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost-devel icu-devel libbitcoin-blockchain-devel
+ libbitcoin-consensus-devel libbitcoin-database-devel libbitcoin-network-devel
+ libbitcoin-node-devel libbitcoin-protocol-devel libbitcoin-system-devel
+ libpng-devel libsecp256k1-devel qrencode-devel zeromq-devel"
+short_desc="Bitcoin Full Node and Query Server"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin-server"
+distfiles="https://codeload.github.com/libbitcoin/$pkgname/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=37ef8d572fb7400565655501ffdea5d07a1de10f3d9fa823d33e2bf68ef8c3ce
+shlib_provides="libbitcoin-server.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-server-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
