diff --git a/common/shlibs b/common/shlibs
index fef5b85e4e..2bb56075b8 100644
--- a/common/shlibs
+++ b/common/shlibs
@@ -3403,3 +3403,4 @@ libglslang.so glslang-6.2.2596_1
 libSPIRV.so glslang-6.2.2596_1
 libsecp256k1.so.0 libsecp256k1-20180531_1
 libbitcoin.so.0 libbitcoin-system-3.5.0_1
+libbitcoin-consensus.so.0 libbitcoin-consensus-3.5.0_1
diff --git a/srcpkgs/libbitcoin-consensus-devel b/srcpkgs/libbitcoin-consensus-devel
new file mode 120000
index 0000000000..01eb2cda87
--- /dev/null
+++ b/srcpkgs/libbitcoin-consensus-devel
@@ -0,0 +1 @@
+libbitcoin-consensus
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-consensus/template b/srcpkgs/libbitcoin-consensus/template
new file mode 100644
index 0000000000..6d801b433c
--- /dev/null
+++ b/srcpkgs/libbitcoin-consensus/template
@@ -0,0 +1,40 @@
+# Template file for 'libbitcoin-consensus'
+pkgname=libbitcoin-consensus
+version=3.5.0
+revision=1
+build_style=gnu-configure
+configure_args="--with-gnu-ld"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost boost-devel libsecp256k1-devel"
+short_desc="Bitcoin Consensus Library"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin-consensus"
+distfiles="https://codeload.github.com/libbitcoin/$pkgname/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=bb29761d4275a9c993151707557008b23572a3d9adecc0e36a3075cfb101dd1e
+shlib_provides="libbitcoin-consensus.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-consensus-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
