diff --git a/common/shlibs b/common/shlibs
index c32025f5ed..3cd91ba962 100644
--- a/common/shlibs
+++ b/common/shlibs
@@ -3406,3 +3406,4 @@ libbitcoin.so.0 libbitcoin-system-3.5.0_1
 libbitcoin-consensus.so.0 libbitcoin-consensus-3.5.0_1
 libbitcoin-protocol.so.0 libbitcoin-protocol-3.5.0_1
 libbitcoin-client.so.0 libbitcoin-client-3.5.0_1
+libbitcoin-network.so.0 libbitcoin-network-3.5.0_1
diff --git a/srcpkgs/libbitcoin-network-devel b/srcpkgs/libbitcoin-network-devel
new file mode 120000
index 0000000000..c464411425
--- /dev/null
+++ b/srcpkgs/libbitcoin-network-devel
@@ -0,0 +1 @@
+libbitcoin-network
\ No newline at end of file
diff --git a/srcpkgs/libbitcoin-network/template b/srcpkgs/libbitcoin-network/template
new file mode 100644
index 0000000000..4f278087ca
--- /dev/null
+++ b/srcpkgs/libbitcoin-network/template
@@ -0,0 +1,41 @@
+# Template file for 'libbitcoin-network'
+pkgname=libbitcoin-network
+version=3.5.0
+revision=1
+build_style=gnu-configure
+configure_args="--with-gnu-ld"
+hostmakedepends="autoconf automake git libtool m4 pkg-config"
+makedepends="boost-devel icu-devel libbitcoin-system-devel libpng-devel
+ libsecp256k1-devel qrencode-devel"
+short_desc="Bitcoin P2P Network Library"
+maintainer="Andy Weidenbaum <archbaum@gmail.com>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/libbitcoin/libbitcoin-network"
+distfiles="https://codeload.github.com/libbitcoin/$pkgname/tar.gz/v$version>$pkgname-$version.tar.gz"
+checksum=e065bd95f64ad5d7b0f882e8759f6b0f81a5fb08f7e971d80f3592a1b5aa8db4
+shlib_provides="libbitcoin-network.so.0"
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=$XBPS_CROSS_TRIPLET --with-boost=$XBPS_CROSS_BASE/usr
+ --with-boost-libdir=$XBPS_CROSS_BASE/usr/lib"
+fi
+
+pre_configure() {
+	./autogen.sh
+}
+
+post_install() {
+	vlicense COPYING
+	vdoc README.md
+}
+
+libbitcoin-network-devel_package() {
+	short_desc+=" - development files"
+	depends="$sourcepkg>=${version}_$revision"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/lib/pkgconfig
+	}
+}
