From 934ced6fa2384584428a7f303e095cb763d6ed8f Mon Sep 17 00:00:00 2001
From: Andy Weidenbaum <atweiden@ioiojo.com>
Date: Mon, 21 Aug 2023 08:11:43 +0000
Subject: [PATCH] allow other CSS class names to precede 'mediawiki'

---
 wiki_lite.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wiki_lite.py b/wiki_lite.py
index f54690e..fcc968d 100644
--- a/wiki_lite.py
+++ b/wiki_lite.py
@@ -98,7 +98,7 @@ def remove_tags(html):
     flags = re.DOTALL | re.MULTILINE
     #body = re.compile('^<body class.*?^NewPP limit report$', flags)
     #body = re.compile('^<body class="mediawiki.*?"site":"loading","user":"ready","user.groups":"ready"', flags)
-    body = re.compile('^<body class="mediawiki.*?<li id="footer-info-copyright">', flags)
+    body = re.compile('^<body class=".*mediawiki.*?<li id="footer-info-copyright">', flags)
     html = body.findall(html)[0]
     html = '\n'.join(html.split('\n')[:-1])
     toc  = re.compile('^<div class="toc noprint" style="text-align: center; margin-bottom: 1em">.*?</div>$', flags)
-- 
2.41.0

